routes/user.js

```diff
@@ -1,18 +1,21 @@
-const express = require('express')
-
-const idValidation = require('~/middlewares/idValidation')
-const asyncWrapper = require('~/middlewares/asyncWrapper')
-const userController = require('~/controllers/user')
-const reviewRouter = require('~/routes/review')
-
-const router = express.Router()
-
-router.param('id', idValidation)
-
-router.use('/:id/:role/reviews', reviewRouter)
-
-router.get('/', asyncWrapper(userController.getUsers))
-router.get('/:id/:role', asyncWrapper(userController.getOneUser))
-router.delete('/:id', asyncWrapper(userController.deleteUser))
-
-module.exports = router
+const express = require('express')
+
+const idValidation = require('~/middlewares/idValidation')
+const asyncWrapper = require('~/middlewares/asyncWrapper')
+const userController = require('~/controllers/user')
+const reviewRouter = require('~/routes/review')
+const { restrictTo } = require('~/middlewares/auth')
+const { roles } = require('~/consts/auth')
+
+const router = express.Router()
+
+router.param('id', idValidation)
+
+router.use('/:id/:role/reviews', reviewRouter)
+
+router.get('/', asyncWrapper(userController.getUsers))
+router.get('/:id/:role', asyncWrapper(userController.getOneUser))
+router.patch('/:id', restrictTo(roles.ADMIN), asyncWrapper(userController.updateUser))

```

Comment:
can use this middleware 'globally' for this route, since delete will be avail for admin only 

router.use(restrictTo(ADMIN))

Reply:
i mean it must be after line 17

Reply:
added restrictTo middleware for role checking

Reply:
also it couldn't get around without authMiddleware and i have added it

Reply:
this middleware restrictTo was a little bit broken, cause there was no next() in case if user is admin and it's no need to throw error